Live audit data

djarotech opened the pull request:
this PR is small. I changed the rc.yaml so that I can do live data . Tested on a gcp instance

devdattakulkarni commented:
Looks fine. Couple of points:

1) Can you squash all the changes in a single commit

2) In the commit message add more context about the change. Have the commit header line a crisp one-line description of change. Then leave a blank line. In the body of the commit message add short description of what the change is, what is its purpose, if there are any special instructions to enable it.

3) I see that you have tested it on GCP. So in a real cluster this will work. What about Minikube though? That will be the first place anyone want to try. Can you investigate bit around what would it take to use live data on Minikube?

devdattakulkarni commented on artifacts/simple-image/Dockerfile:
```diff
@@ -14,4 +14,8 @@
 
 FROM fedora
 ADD kube-provenance-apiserver /
+<<<<<<< HEAD
```
Stray line coming probably coming from merge conflict.
Should be removed.

devdattakulkarni commented on artifacts/simple-image/Dockerfile:
```diff
@@ -14,4 +14,8 @@
+<<<<<<< HEAD
+ADD minikube-sample-audit.log /tmp/
+=======
+>>>>>>> 258a45ce55e9cde9247d86289d08bb80090ca50f
```
Same comment as above for lines 19 and 20.

devdattakulkarni commented on artifacts/example/rc.yaml:
```diff
@@ -37,5 +41,4 @@ spec:
             path: /tmp/kube-apiserver-audit.log
-            # this field is optional
-            type: File
+            type: FileOrCreate
```
+1

devdattakulkarni commented on pkg/provenance/provenance.go:
```diff
@@ -73,14 +75,30 @@ func init() {
+func onMinikube() bool {
+	ip := os.Getenv("HOST_IP")
+	ipBeginsWith := strings.Split(ip, ".")[0]
+	return ipBeginsWith == "10"
```
Minikube IPs also begin with 198. for v0.25 and below.
That should be accommodated here as well.
You can have a OR condition for these two cases.

devdattakulkarni commented:
Check my in-line comments.

Also, two other comments.

1) Can you squash all the commits into a single commit? Check how to do it here:

https://stackoverflow.com/questions/14534397/squash-all-my-commits-into-one-for-github-pull-request

https://gist.github.com/patik/b8a9dc5cd356f9f6f980

Once you have squashed the commits you will have to force push the branch as shown on the second link.

2) Make the commit message informative. Explain about two options being supported (Minikube and real setup). Point out that on Minikube live data is not being used.

djarotech commented:
Done! please review

devdattakulkarni commented on pkg/provenance/provenance.go:
```diff
@@ -420,59 +429,63 @@ func handleCompositeFields(vSliceMap []map[string]string, mapRelationships map[s
 	return false
 }
+//Method that returns true if all the elements in boolSlice are True
+func all(boolSlice []bool) bool{
```
This change seems not related to Audit logs.

djarotech replied:
How strange . This method was merged in the last pull request, the left hand side (master) looks incorrect  .

devdattakulkarni commented:
Looks good after squashing. Few more points related to commit message:

Top line: Use following
"Support for live audit data (and misc. changes)"

Body: Include following two points:
- Current top-line + body (combine them)
- Miscellaneous formatting changes.

djarotech committed:
Support for live audit data (and misc changes)

In order to see live changes to the kube-apiserver-audit.log file, I mounted a volume with the File that is constantly being updates by Kubernetes, upon changes to a custom resource. Furthermore, to support live audit data, I made changes so that the project can be run on minikube. However, since minikube does not officially support audit logging, I am using a sample minikube audit log  that I have generated using a Postgres custom resource and applying a few functions to generate the log. Now, the project continuously rebuilds its provenance history when it is on Kubernetes but not minikube since auditing is not supported(it doesn'nt need torebuild). To check if I am on minikube or kubernetes I use status.hostIP in rc.yaml.

Commit:
Merge branch 'master' into live-audit-data
